src/main/java/gyro/azure/CloudBlobFileBackend.java

```
@@ -0,0 +1,91 @@
+package gyro.azure;
+
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import gyro.azure.storage.StorageAccountResource;
+import gyro.core.FileBackend;
+import gyro.core.GyroException;
+import gyro.core.Type;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.stream.Stream;
+
+@Type("cloudblob")
+public class CloudBlobFileBackend extends FileBackend {
+    private String storageAccount;
+    private String cloudBlobContainer;
+    private String prefix;
+
+    public String getStorageAccount() {
+        return storageAccount;
+    }
+
+    public void setStorageAccount(String storageAccount) {
+        this.storageAccount = storageAccount;
+    }
+
+    public String getCloudBlobContainer() {
+        return cloudBlobContainer;
+    }
+
+    public void setCloudBlobContainer(String cloudBlobContainer) {
+        this.cloudBlobContainer = cloudBlobContainer;
+    }
+
+    public String getPrefix() {
+        return prefix;
+    }
+
+    public void setPrefix(String prefix) {
+        this.prefix = prefix;
+    }
+
+    @Override
+    public Stream<String> list() throws Exception {
+        return null;
+    }
+
+    @Override
+    public InputStream openInput(String file) throws Exception {
+        return container().getBlockBlobReference(prefixed(file)).openInputStream();
+    }
+
+    @Override
+    public OutputStream openOutput(String file) throws Exception {
+        return new ByteArrayOutputStream() {
+            public void close() {
+                try {
+                    container().getBlockBlobReference(prefixed(file)).uploadFromByteArray(toByteArray(), 0, toByteArray().length);
+                } catch (Exception e) {
+
```

This should throw an exception.